config/vlan.py

```diff
@@ -189,97 +231,159 @@ def vlan_member():
+@click.option('-m', '--multiple', is_flag=True, help="Add Multiple Vlan(s) in Range or in Comma separated list")
+@click.option('-e', '--except_flag', is_flag=True, help="Skips the given vlans and adds all other existing vlans")
 @clicommon.pass_db
-def del_vlan_member(db, vid, port):
+def del_vlan_member(db, vid, port, multiple, except_flag):
     """Delete VLAN member"""
 
     ctx = click.get_current_context()
-    log.log_info("'vlan member del {} {}' executing...".format(vid, port))
-    vlan = 'Vlan{}'.format(vid)
     
+    # parser will parse the vid input if there are syntax errors it will throw error
+
+    vid_list = clicommon.vlan_member_input_parser(ctx,"del", db, except_flag, multiple, vid, port)
+
     config_db = ValidatedConfigDBConnector(db.cfgdb)
     if ADHOC_VALIDATION:
-        if not clicommon.is_vlanid_in_range(vid):
-            ctx.fail("Invalid VLAN ID {} (1-4094)".format(vid))
-
-        if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
-            ctx.fail("{} does not exist".format(vlan))
-
-        if clicommon.get_interface_naming_mode() == "alias": # TODO: MISSING CONSTRAINT IN YANG MODEL
-            alias = port
-            iface_alias_converter = clicommon.InterfaceAliasConverter(db)
-            port = iface_alias_converter.alias_to_name(alias)
-            if port is None:
-                ctx.fail("cannot find port name for alias {}".format(alias))
-
-        if not clicommon.is_port_vlan_member(db.cfgdb, port, vlan): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is not a member of {}".format(port, vlan))
-
-    try:
-        config_db.set_entry('VLAN_MEMBER', (vlan, port), None)
-        delete_db_entry("DHCPv6_COUNTER_TABLE|{}".format(port), db.db, db.db.STATE_DB)
-        delete_db_entry("DHCP_COUNTER_TABLE|{}".format(port), db.db, db.db.STATE_DB)
-    except JsonPatchConflict:
-        ctx.fail("{} invalid or does not exist, or {} is not a member of {}".format(vlan, port, vlan))
+        for vid in vid_list:
+
+            log.log_info("'vlan member del {} {}' executing...".format(vid, port))
+
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+
+            vlan = 'Vlan{}'.format(vid)
+
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+                log.log_info("{} does not exist".format(vlan))
+                ctx.fail("{} does not exist, Aborting!!!".format(vlan))
+
+            if clicommon.get_interface_naming_mode() == "alias":  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                alias = port
+                iface_alias_converter = clicommon.InterfaceAliasConverter(db)
+                port = iface_alias_converter.alias_to_name(alias)
+                if port is None:
+                    ctx.fail("cannot find port name for alias {}".format(alias))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
```

Comment:
same as above

Reply:
Please note that these are exisiting YANG Constraints and our proposed changes/YANG Model has not modified any of these constraints.